sympy/utilities/iterables.py

```diff
@@ -523,7 +521,18 @@ def ibin(n, bits=0, str=False):
     ['000', '001', '010', '011', '100', '101', '110', '111']
 
     """
+    if n < 0:
+        raise ValueError(
+                "Negetive numbers are not allowed")
+    x, y = abs(n), []
```

Comment:
no need for new code here...the routine already works and we are only testing for two conditions at the start.

Reply:
I have removed the x, y condition but kept the n<0 one.